pkg/apps/application-list.jsx

```diff
@@ -119,13 +124,35 @@ export const ApplicationList = ({ metainfo_db, appProgress, appProgressTitle, ac
         }
     }
 
+    const check_missing_data = useCallback(() => {
+        if (dataPackage !== null) {
+            setDataPackagesInstalled(null);
+            check_missing_packages(dataPackage)
+                    .then(data => setDataPackagesInstalled(data.missing_names.length === 0 && data.unavailable_names.length === 0));
```

Comment:
The [test](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19281-20230908-155558-64d3912b-fedora-38-other/log.html#175) is crashing with `RuntimeError: TypeError: Cannot read properties of null (reading 'length')`. I suggest to use `?.length` in both cases here.

Reply:
Makes sense, done